embed/etcd.go

```diff
@@ -205,6 +208,11 @@ func (e *Etcd) Close() {
 			e.Clients[i].Close()
 		}
 	}
+	for i := range e.metricsListeners {
+		if e.metricsListeners[i] != nil {
```

Comment:
no need for the nil check? metricsListeners is only appended with non-nil listeners